sdk/program/src/loader_upgradeable_instruction.rs

```diff
@@ -0,0 +1,75 @@
+//! Instruction types
+
+use crate::pubkey::Pubkey;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum UpgradeableLoaderInstruction {
+    /// Write program data into a source account.
+    ///
+    /// A source account is an intermediary that once fully populated is used
+    /// with the `Finalize` instruction to populate the program's code account.
+    ///
+    /// # Account references
+    ///   0. [writable, signer] source account to write program data to.
+    Write {
```

Comment:
How will `Write` know `0` is a source account and not a code or executable account?

Reply:
Good point, will need an `initialize_source_account` instruction